### ISSUE TITLE
`File.open` ignores `flags` option

### ISSUE THREAD

**Issue by mhib** [oracle/truffleruby#2820]
Code to reproduce the issue:
```ruby
File.open("not_existing_file", flags: File::CREAT | File::WRONLY)
````

This line returns a `File` both on CRuby and JRuby, but throws an error on TruffleRuby.

My ruby --version:
```
truffleruby 22.3.0, like ruby 3.0.3, GraalVM CE Native [x86_64-darwin]
```

**Comment by eregon** [comment]
Thank you for the report.
Could you share the stracktrace/error? (or part of it)

**Comment by eregon** [comment]
I can repro on truffleruby0dev:
```
$ ruby -e 'File.open("not_existing_file", flags: File::CREAT | File::WRONLY)'
<internal:core> core/errno.rb:48:in `handle': No such file or directory - not_existing_file (Errno::ENOENT)
	from <internal:core> core/io.rb:898:in `sysopen'
	from <internal:core> core/file.rb:1216:in `initialize'
	from <internal:core> core/io.rb:674:in `open'
	from -e:1:in `<main>'
```

**Comment by andrykonchin** [comment]
Fixed in 8bb04d8cda1487dd61bb14bed3722f51de24bb91
